### PR TITLE
redesign: allow sharding key to be undefined or null

### PR DESCRIPTION
**Pull request by william-silversmith**

### CODE REVIEW

File: src/neuroglancer/datasource/precomputed/frontend.ts [hunk 1]
```diff
@@ -242,7 +242,7 @@ function parseShardingEncoding(y: any): DataEncoding {
 }
 
 function parseShardingParameters(shardingData: any): ShardingParameters|undefined {
-  if (shardingData === undefined) return undefined;
+  if (shardingData === undefined || shardingData === null) return undefined;
```
Comment (ogewan) [inline src/neuroglancer/datasource/precomputed/frontend.ts]: if (!shardingData) return;
is equivalent

Comment (ogewan) [reply src/neuroglancer/datasource/precomputed/frontend.ts]: It's good enough as is tbh, if you don't feel like changing it I'll merge it in.

Comment (william-silversmith) [reply src/neuroglancer/datasource/precomputed/frontend.ts]: Not exactly, that would also count `0, '', false, NaN` which would be more likely to trigger an exception right after.

Comment (william-silversmith) [reply src/neuroglancer/datasource/precomputed/frontend.ts]: Would it be okay to merge this version? Thanks!

**Comment by ogewan** [comment]
lgtm

**Pull request merged**